api/app/Actions/EventType/CloneEventTypeByIdAction.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\EventType;
+
+use App\Exceptions\EventType\NameIsAlreadyInUseException;
+use App\Exceptions\EventType\SlugIsAlreadyInUseException;
+use App\Repositories\EventType\Criterion\NameCriterion;
+use App\Repositories\EventType\Criterion\SlugCriterion;
+use App\Repositories\EventType\EventTypeRepositoryInterface;
+use Illuminate\Support\Str;
+
+final class CloneEventTypeByIdAction
+{
+    private EventTypeRepositoryInterface $eventTypeRepository;
+    private const MAX_NAME_LENGTH = 50;
+
+    public function __construct(EventTypeRepositoryInterface $eventTypeRepository)
+    {
+        $this->eventTypeRepository = $eventTypeRepository;
+    }
+
+    public function execute(CloneEventTypeByIdRequest $request): CloneEventTypeByIdResponse
+    {
+        $eventType = $this->eventTypeRepository->getById($request->getId());
+        $newEventType = $eventType->replicate();
+
+        $newSlug = $newEventType->slug . '-copy';
+        $newSlug = substr_replace($newSlug, Str::random(2), -1);
```

Comment:
Why not simply add a random string along with adding `-copy` suffix at once?

Reply:
I think that these random characters will increase our slug length so much